### PR TITLE
Branding and required changes

### PR DESCRIPTION
**Pull request by jniederm**
https://github.com/oVirt/ovirt-web-ui/wiki/Branding

### CODE REVIEW

File: ovirt-web-ui.spec.in [hunk 1]
```diff
@@ -73,7 +78,7 @@ make install DESTDIR=%{buildroot}
 * Mon May 15 2017 Marek Libra <mlibra@redhat.com> - 0.2.1
 - React and Redux updated to latest version
 * Fri May 12 2017 Marek Libra <mlibra@redhat.com> - 0.2.0
-- Add/Edit VM Dialog
+- Add/Edit VM Dialogls
```
Comment (bond95) [inline ovirt-web-ui.spec.in]: "Dialogs"

File: package.json [hunk 2]
```diff
@@ -38,7 +38,7 @@
     "file-loader": "0.9.0",
     "filesize": "3.3.0",
     "find-cache-dir": "0.1.1",
-    "flow-babel-webpack-plugin": "1.0.1",
+    "flow-babel-webpack-plugin": "^1.1.0",
```
Comment (bond95) [inline package.json]: Remove ^ ?

File: configure.ac [hunk 3]
```diff
@@ -18,7 +18,7 @@ AC_PREREQ(2.60)
 
 define([VERSION_MAJOR], [1])
 define([VERSION_MINOR], [1])
-define([VERSION_FIX], [0])
+define([VERSION_FIX], [1])
```
Comment (mareklibra) [inline configure.ac]: Please do not increase version within this PR.

New ovirt-web-ui version will be released independently.

File: ovirt-web-ui.spec.in [hunk 4]
```diff
@@ -53,8 +53,11 @@ make install DESTDIR=%{buildroot}
 %{_ovirtwebuidir}
 %{_datarootdir}/ovirt-engine/ovirt-web-ui.war
 %{_sysconfdir}/ovirt-engine/engine.conf.d/50-ovirt-web-ui.conf
+%{_sysconfdir}/ovirt-web-ui/branding/00-ovirt.brand
 
 %changelog
+* Tue Jul 18 2017 Jakub Niedermertl <jniederm@redhat.com> - 1.1.1
```
Comment (mareklibra) [inline ovirt-web-ui.spec.in]: Proper release will be done later, following more general release process.

File: ovirt-web-ui.spec.in (hunk 1 above)
Comment (mareklibra) [inline ovirt-web-ui.spec.in]: version 0.2.0 has been already released.
Please do not modify this.

File: package.json (hunk 2 above)
Comment (mareklibra) [inline package.json]: Yes, please.

File: src/version.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 const Product = {
   name: 'oVirt Basic Portal',
-  version: '1.1.0',
```
Comment (mareklibra) [inline src/version.js]: This file does not need to be modified.

It is regenerated by ./autogen.sh according to configure.ac.

File: ovirt-web-ui.spec.in (hunk 1 above)
Comment (jniederm) [inline ovirt-web-ui.spec.in]: done

Comment (jniederm) [inline ovirt-web-ui.spec.in]: Done

File: package.json (hunk 2 above)
Comment (jniederm) [inline package.json]: Done

File: configure.ac (hunk 3 above)
Comment (jniederm) [inline configure.ac]: done

File: ovirt-web-ui.spec.in (hunk 4 above)
Comment (jniederm) [inline ovirt-web-ui.spec.in]: done

File: src/version.js (hunk 5 above)
Comment (jniederm) [inline src/version.js]: done

**Comment by jniederm** [comment]
Hi Marek, why to remove `yarn.lock` changes out of pull?

I find changes in package.json dependencies and yarn.lock to be tightly coupled. If one checkouts a commit containing package.json changes but not yarn.lock changes then local yarn dependency resolution might result in different versions (with respect to versions computed on machine of commit author) (due to new releases of dependencies) and consequently in different application behavior which I find undesirable.

**Comment by mareklibra** [comment]
@jniederm , right, they are tightly coupled. I think it's rare to checkout just the exact commit. Anyway, the commit right after is with the one with updated yarn.lock. Developers usually checkout master.

The reason for this splitting is simple: merging might become difficult. The `yarn.lock` is generated and it's quite complex. Irregularities can't be find manually at this state. If a buggy merging passes, it will usually appear during production build or when testing rpm result. Means at time pressure.

Since usually I'm the one who recently does merging, I'm fine to commit myself to update `yarn.lock` right after merge. This way, it's trivial for me to ensure, the yarn.lock is not corrupted.

Btw. if such a yarn.lock merge introduces a mess (might happen, we are humans), i.e. build of ovirt-nodejs-modules will fail with consequences to multiple other projects.

**Comment by jniederm** [comment]
I see, `yarn.lock` removed.

File: packaging/ovirt-web-ui.war/WEB-INF/branding.jsp [hunk 6]
```diff
@@ -0,0 +1,129 @@
+<%@ page language="java" contentType="text/html; charset=UTF-8" pageEncoding="UTF-8"%>
```
Comment (mareklibra) [inline packaging/ovirt-web-ui.war/WEB-INF/branding.jsp]: Can we move this .jsp file one level higher? Among other `.jsp` files.

File: Makefile.am [hunk 7]
```diff
@@ -91,15 +92,18 @@ ovirt-web-ui:
...
 	$(MKDIR_P) $(DESTDIR)$(OVIRT_ENGINE_DIR)
+	$(MKDIR_P) $(DESTDIR)$(sysconfdir)/ovirt-web-ui/branding
 	cp -rpv build/ovirt-web-ui.war $(DESTDIR)$(USER_PORTAL_DIR)
-	cp -rpv build/etc $(DESTDIR)/$(sysconfdir)
+	cp -rpvT build/branding $(DESTDIR)$(USER_PORTAL_DIR)/branding/ovirt
+	cp -rpv build/etc/* $(DESTDIR)$(sysconfdir)
 	echo "ln -s" $(DESTDIR)$(USER_PORTAL_DIR)/ovirt-web-ui.war $(DESTDIR)$(OVIRT_ENGINE_DIR)/ovirt-web-ui.war
-	ln -s ../ovirt-web-ui/ovirt-web-ui.war $(DESTDIR)$(OVIRT_ENGINE_DIR)/ovirt-web-ui.war
+	ln -sf ../ovirt-web-ui/ovirt-web-ui.war $(DESTDIR)$(OVIRT_ENGINE_DIR)/ovirt-web-ui.war
+	ln -sfr $(DESTDIR)$(USER_PORTAL_DIR)/branding $(DESTDIR)$(sysconfdir)/ovirt-web-ui/branding/00-ovirt.brand
```
Comment (mareklibra) [inline Makefile.am]: For production environment this shall be:

ln -sfr $(DESTDIR)$(USER_PORTAL_DIR)/branding/**ovirt** $(DESTDIR)$(sysconfdir)/ovirt-web-ui/branding/00-ovirt.brand

Comment (mareklibra) [inline Makefile.am]: Development environment is still working with this change.

File: src/branding.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
+// @flow
```
Comment (mareklibra) [inline src/branding.js]: Is it just my observation or the `flow` make the code less readable?

File: Makefile.am [hunk 9]
```diff
@@ -25,14 +25,15 @@ EXTRA_DIST = \
...
+	branding \
 	src \
 	config \
 	doc \
 	scripts \
 	packaging \
 	package.json \
 	index.html \
-  index.jsp \
+    index.jsp \
```
Comment (jniederm) [inline Makefile.am]: Absolutely. And do you disagree with it in this pull or in general? I mean it looks like an indentation fix to me.

File: Makefile.am (hunk 7 above)
Comment (jniederm) [inline Makefile.am]: Thanks, it stopped working for me an I wondered why.

File: src/branding.js (hunk 8 above)
Comment (jniederm) [inline src/branding.js]: I had the same experience, and then I got used to that. I makes the source to look like a different language.

File: packaging/ovirt-web-ui.war/WEB-INF/branding.jsp (hunk 6 above)
Comment (jniederm) [inline packaging/ovirt-web-ui.war/WEB-INF/branding.jsp]: WEB-INF allows to hide files form direct access. If we move it one level up, there would be a need to come up with something smart to return for `GET /web-ui/branding.jsp` and additional unnecessary code in `branging.jsp`. This way only `/branding/*` is handled. So I'd rather keep it here.

Comment (mareklibra) [inline packaging/ovirt-web-ui.war/WEB-INF/branding.jsp]: ok, makes sense

**Pull request merged**